packages/field-label/src/FieldLabel.ts

```diff
@@ -101,9 +101,11 @@ export class FieldLabel extends SizedMixin(SpectrumElement) {
             const applyLabel = this.target.applyFocusElementLabel;
             const focusable = this.target.focusElement || this.target;
             const targetParent = focusable.getRootNode() as HTMLElement;
+
             if (typeof applyLabel !== 'undefined') {
                 applyLabel(this.labelText);
-            } else if (targetParent === (this.getRootNode() as HTMLElement)) {
+            }
+            if (targetParent === (this.getRootNode() as HTMLElement)) {
```

Comment:
While I admire the simplicity of this change, allowing multiple paths to be taken in this logic can lead to unexpected outcomes.

In specific, while it is not explicitly negative, you can see when comparing this branch to `main` that when this is leveraged [in conjunction with `<sp-picker>`](https://issue-number-field-labels--spectrum-web-components.netlify.app/storybook/?path=/story/picker--default) this change applies an `aria-label` to the `<button>` within the Picker's shadow root, that is not currently being applied in `main`. 

I _think_ the path would be to either imperatively apply `aria-label` to the `<input>` in the Number Field as part of its `applyFocusElementLabel` method, or to hand state in the root Textfield component that could be adjusted to accept this value into its `render` method.